src/arch/aarch64/CodeGen.zig

```diff
@@ -6149,22 +6144,7 @@ fn resolveInst(self: *Self, inst: Air.Inst.Ref) InnerError!MCValue {
         .val = (try self.air.value(inst, mod)).?,
     });
 
-    switch (self.air.instructions.items(.tag)[inst_index]) {
-        .interned => {
-            // Constants have static lifetimes, so they are always memoized in the outer most table.
-            const branch = &self.branch_stack.items[0];
-            const gop = try branch.inst_table.getOrPut(self.gpa, inst_index);
-            if (!gop.found_existing) {
-                const interned = self.air.instructions.items(.data)[inst_index].interned;
-                gop.value_ptr.* = try self.genTypedValue(.{
-                    .ty = inst_ty,
-                    .val = interned.toValue(),
-                });
-            }
-            return gop.value_ptr.*;
-        },
-        else => return self.getResolvedInstValue(inst_index),
-    }
+    return self.getResolvedInstValue(inst_index);
```

Comment:
oh, this is a lovely diff

Reply:
Yeah it turns out having the separation be "instructions vs values" instead of "instructions vs a few special values vs instructions which represent values" cleans a few things up